src/direct-auth-idx/direct-auth-idx/Controllers/InteractionCodeController.cs

```diff
@@ -0,0 +1,102 @@
+{
+    public class InteractionCodeController : Controller
+    {
+        private readonly IAuthenticationManager _authenticationManager;
+        private readonly IIdxClient _idxClient;
+
+        public InteractionCodeController(IAuthenticationManager authenticationManager, IIdxClient idxClient)
+        {
+            this._authenticationManager = authenticationManager;
+            this._idxClient = idxClient;
+        }
+
+        /// <summary>
+        /// This method is called by a 302 browser redirect back to the application from Okta after successful authentication to an external
+        /// identity provider.
+        /// </summary>
+        /// <param name="state">The state handle.</param>
+        /// <param name="interaction_code">The interaction code.  This is the value that is exchanged for tokens.</param>
+        /// <param name="error">The error if an error occurred.</param>
+        /// <param name="error_description">The error description if an error occurred.</param>
+        /// <returns></returns>
+        public async Task<ActionResult> Callback(string state = null, string interaction_code = null, string error = null, string error_description = null)
+        {
+            IIdxContext idxContext = Session[state] as IIdxContext;
+
+            if("interaction_required".Equals(error))
+            {
+                return View("Error", new InteractionCodeErrorViewModel { Error = error, ErrorDescription = "Multifactor Authentication and Social Identity Providers is not currently supported, Authentication failed." });
+            }
+
+            if (!string.IsNullOrEmpty(error))
+            {
+                return View("Error", new InteractionCodeErrorViewModel { Error = error, ErrorDescription = error_description });
+            }
+
+            if (string.IsNullOrEmpty(interaction_code))
+            {
+                return View("Error", new InteractionCodeErrorViewModel { Error = "null_interaction_code", ErrorDescription = "interaction_code was not specified" });
+            }
+
+            await RedeemInteractionCodeAndSignInAsync(idxContext, interaction_code);
+            return RedirectToAction("Index", "Home");
+        }
+
+        private async Task RedeemInteractionCodeAndSignInAsync(IIdxContext idxContext, string interactionCode)
+        {
+            try
+            {
+                Okta.Idx.Sdk.TokenResponse tokens = await _idxClient.RedeemInteractionCodeAsync(idxContext, interactionCode);
+
+                if (tokens == null)
+                {
+                    HttpContext.Response.Redirect("/");
+                }
+                else
+                {
+                    ClaimsIdentity claimsIdentity = await GetClaimsIdentityAsync(tokens);
+                    _authenticationManager.SignIn(new AuthenticationProperties { IsPersistent = false }, claimsIdentity);
+                }
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
```

Comment:
If there's an exception, we're currently showing the error message on the same page. Can we do the same here for consistency? https://github.com/okta/okta-idx-dotnet/blob/idx-auth-layer/src/direct-auth-idx/direct-auth-idx/Controllers/AccountController.cs#L76